File: compiler/testData/foreignAnnotationsJava8/tests/jspecify/java/IgnoreAnnotations.java
import org.jspecify.annotations.*;

@DefaultNonNull
public class IgnoreAnnotations {
    @Nullable public Derived field = null;

    @Nullable
    public Derived foo(Derived x, @NullnessUnspecified Base y) {
        return null;
    }

    public Derived everythingNotNullable(Derived x) { return null; }

    public @Nullable Derived everythingNullable(@Nullable Derived x) { return null; }

    public @NullnessUnspecified Derived everythingUnknown(@NullnessUnspecified Derived x) { return null; }
}

class Base {
    void foo() {}
}

class Derived extends Base { }

class Use {
    static void main(IgnoreAnnotations a, Derived x) {
        a.foo(x, null).foo();
        // jspecify_nullness_mismatch
        a.foo(null, x).foo();

        a.field.foo();

        // jspecify_nullness_mismatch
        a.everythingNotNullable(null).foo();
        a.everythingNotNullable(x).foo();

        a.everythingNullable(null).foo();

        a.everythingUnknown(null).foo();
    }
}

File: compiler/testData/foreignAnnotationsJava8/tests/jspecify/ignoreAnnotations.kt
// !DIAGNOSTICS: -UNUSED_VARIABLE -UNUSED_PARAMETER
// JSPECIFY_STATE ignore
// FILE: A.java

import org.jspecify.annotations.*;

public class A {
    @Nullable public String field = null;

    @Nullable
    public String foo(@NotNull String x, @NullnessUnspecified CharSequence y) {
        return "";
    }

    @NotNull
    public String bar() {
        return "";
    }

    @DefaultNonNull
    public String everythingNotNullable(String x) { return ""; }

    @DefaultNullable
    public String everythingNullable(String x) { return ""; }

    @DefaultNullnessUnspecified
    public String everythingUnknown(String x) { return ""; }
}

// FILE: main.kt

fun main(a: A) {
    a.foo("", null)?.length
    a.foo("", null).length
    a.foo(null, "").length

    a.bar().length
    a.bar()!!.length

    a.field?.length
    a.field.length

    a.everythingNotNullable(null)?.length
    a.everythingNotNullable(null).length
    a.everythingNotNullable("").length

    a.everythingNullable(null).length
    a.everythingNullable(null)?.length

    a.everythingUnknown(null).length
    a.everythingUnknown(null)?.length
}
